packages/openapi-to-graphql/src/oas_3_tools.ts

```diff
@@ -1026,64 +1026,78 @@ export function getSecurityRequirements(
   return results
 }
 
+export enum CaseStyle {
+  PascalCase, // Used for type names
+  camelCase, // Used for (input) object field names
+  ALL_CAPS // Used for enum values
+}
+
 /**
  * First sanitizes given string and then also camel-cases it.
  */
-export function sanitize(
-  str: string,
-  lowercaseFirstChar: boolean = true
-): string {
+export function sanitize(str: string, caseStyle: CaseStyle): string {
   /**
    * Remove all GraphQL unsafe characters
-   *
-   * Also remove _, even though it is GraphQL safe, to follow prescribed
-   * GraphQL naming conventions, which either use PascalCase, camelCase,
-   * or ALL_CAPS (the last of which has not been implemented)
-   *
-   * TODO: Adapt to allow for ALL_CAPS
    */
-  let sanitized = str.split(/[^a-zA-Z0-9]/g).reduce((path, part) => {
-    return path + capitalize(part)
+  const regex =
+    caseStyle === CaseStyle.ALL_CAPS
+      ? /[^a-zA-Z0-9_]/g // ALL_CAPS has underscores
+      : /[^a-zA-Z0-9]/g
+  let sanitized = str.split(regex).reduce((path, part) => {
+    if (caseStyle === CaseStyle.ALL_CAPS) {
+      return path + '_' + part
+    } else {
+      return path + capitalize(part)
+    }
   })
 
+  switch (caseStyle) {
+    case CaseStyle.PascalCase:
+      // The first character in PascalCase should be uppercase
+      sanitized = capitalize(sanitized)
+      break
+
+    case CaseStyle.camelCase:
+      // The first character in camelCase should be lowercase
+      sanitized = uncapitalize(sanitized)
+      break
+
+    case CaseStyle.ALL_CAPS:
+      // Delete first underscore
+      if (sanitized.charAt(0) === '_') {
+        sanitized = sanitized.substr(0)
+      }
+      sanitized = sanitized.toUpperCase()
+      break
+  }
+
   // Special case: we cannot start with number, and cannot be empty:
   if (/^[0-9]/.test(sanitized) || sanitized === '') {
     sanitized = '_' + sanitized
   }
 
-  // First character should be lowercase
-  if (lowercaseFirstChar) {
-    sanitized = uncapitalize(sanitized)
-  }
-
   return sanitized
 }
 
 /**
  * Sanitizes the given string and stores the sanitized-to-original mapping in
  * the given mapping.
  */
-export function sanitizeAndStore(
+export function storeSaneName(
```

Comment:
This didn't really make sense trying to squash two different operations into the same function.

I think this is also a needed step for #274

Reply:
Jupp, I like separating these two capabilities.